### ISSUE TITLE
bootstrap-conda.sh failing on build (no .bashrc found)

### ISSUE THREAD

**Issue by nehalecky** [GoogleCloudPlatform/dataproc-initialization-actions#41]
In the past 24 hours, our builds (w/ @jeffkpayne) against Google Cloud Dataproc have been failing due to an error in the `./conda/bootstrap-conda.sh` init action script. Targeting this script as the primary init action in a dataproc cluster creation call like:

```bash
> gcloud dataproc clusters create bombora-dev-analytics-test --initialization-actions=gs://bombora-dev-analytics/dataproc-init-actions/conda/bootstrap-conda.sh
```

results in multiple errors on master and worker nodes

```
Waiting on operation [projects/bombora-dev/regions/global/operations/0f85471c-c270-4044-baad-9b38b6ba9194].
Waiting for cluster creation operation...done.
ERROR: (gcloud.dataproc.clusters.create) Operation [projects/bombora-dev/regions/global/operations/0f85471c-c270-4044-baad-9b38b6ba9194] failed: Multiple Errors:
 - Google Cloud Dataproc Agent reports failure. If logs are available, they can be found in 'gs://dataproc-e66c6e3d-80da-4211-b66c-0109bbe4567a-us/google-cloud-dataproc-metainfo/be3ac51a-6e50-449c-9516-5114350408c3/bombora-dev-analytics-test-m'.
 - Google Cloud Dataproc Agent reports failure. If logs are available, they can be found in 'gs://dataproc-e66c6e3d-80da-4211-b66c-0109bbe4567a-us/google-cloud-dataproc-metainfo/be3ac51a-6e50-449c-9516-5114350408c3/bombora-dev-analytics-test-w-1'..
```
Tailing `dataproc-initialization-script-0_output` reveals inability to resolve `/root/.bashrc`?

```
Python 3.5.1 :: Continuum Analytics, Inc.
creating default environment...
installation finished.
Setting environment variables...
Updated PATH: /usr/local/bin/miniconda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
And also HOME:
/usr/local/bin/miniconda/bin/conda
/etc/google-dataproc/startup-scripts/dataproc-initialization-script-0: line 90: /root/.bashrc: No such file or directory
```

As this was previously working via our internal testing (along with @dennishuo testing and accepting PR #18), I'm unsure as to what changes might have occurred? Any tips or suggestions? Thanks much!

**Comment by nehalecky** [comment]
For reference, logs from master node:

- [dataproc-initialization-script-0_output](https://github.com/GoogleCloudPlatform/dataproc-initialization-actions/files/170536/dataproc-initialization-script-0_output.txt)
- [dataproc-startup-script_output](https://github.com/GoogleCloudPlatform/dataproc-initialization-actions/files/170537/dataproc-startup-script_output.txt)

**Comment by nehalecky** [comment]
Was able to resolve simply by removing the sourcing calls to `/root/.bashrc` (no longer needed due to setting PATH env vars and other configs in global profiles). This properly allowed conda to be installed along with all dependencies, and also ensures updated path (exposing miniconda) for both `root` and non-root users. Will submit patch shortly.

**Comment by nehalecky** [comment]
OK, the PR above should resolve this. It has in my testing, including functionality contained in existing PRs depending on conda. Give it a try and we should get this merged asap. Thanks!

**Comment by dennishuo** [comment]
Sorry for the breakage! We indeed rolled out some minor cleanup of the VM environment for newly created clusters, and didn't intend to change anything in a breaking way. We haven't quite standardized the testing of known initialization actions in the release process yet, but in the future we should be able to catch these before any public releases. I'll take a look at the PR.

**Comment by nehalecky** [comment]
@dennishuo, thanks for the quick resolution. We look forward to contributing to the testing of init actions, and I believe a first step would be to expose some google compute engine resources to be for spinning up a minimal dataproc cluster and running a test suite. Internally, we will be implementing this soon, but believe it would be awesome if Google would contribute some resources to do the same.

Also, from our side, is there anything we can do to prevent such breakages in the future (any way to reference specific subminor dataproc versions)? 

Thanks again!

**Comment by dennishuo** [comment]
@nehalecky We're definitely hoping to provide a nice sandboxed CI/test setup for running minimal test suites, unfortunately can't guarantee a timeline at this point since we're still shuffling around various prioritizations.

It's indeed possible to reference specific subminor versions, but the long-term deprecation policies cover major.minor version "tracks" and not specific subminor versions, so they may be subject to shorter lifecycles of being officially supported compared to the [lifecycles of major.minor tracks](https://cloud.google.com/dataproc/versioning#image_version_and_cloud_dataproc_support).

That said, in this case the previous subminor version that we'd expect to still work with the old conda init action would be version "1.0.0", and the current version is "1.0.1". Pinning to 1.0.1 now would ensure workloads are unaffected when 1.0.2 comes out, but it'd be strongly recommended to switch to newer subminor versions as soon as possible; previous subminor versions wouldn't get bug fixes or new features.

We'll probably add a nicer way to query for the list of supported and deprecated subminor versions, but in the meantime you can at least see the complete list by creating a cluster and specifying a bogus version like `--image-version=notavalidversion`

**Comment by nehalecky** [comment]
@dennishuo, thank you for the comprehensive reply. We're excited about these future developments and will continue to work alongside you all to see this advanced! Have a great weekend!
